securedrop_client/gui/widgets.py

```diff
@@ -1084,9 +1084,18 @@ def set_snippet(self, source, uuid=None, content=None):
                 msg_text = content
             else:
                 msg_text = str(msg)
-            if len(msg_text) > 120:
-                msg_text = msg_text[:120] + "..."
-            self.preview.setText(msg_text)
+            if len(msg_text) > 150:
+                msg_text = msg_text[:150] + "..."
+            # Discover word wrap limit.
+            wrap_limit = 0
+            fm = self.preview.fontMetrics()
+            sentence = ""
+            for c in msg_text:
+                sentence += c
+                if fm.horizontalAdvance(sentence) > self.PREVIEW_WIDTH:
```

Comment:
`self.preview` is a `SecureQLabel` so you can use its wordwrapping and elidedText support - that way we don't have to support the same code in multiple places